### DISCUSSION TITLE
Does wire generate Interfaces for Android Java?

### DISCUSSION THREAD

**Discussion by rndhacker** [square/wire#2487] [Q&A]
I was reading the documentation and trying to follow it, it seems like wire does provide some very cool client interfaces to extend and use. Unfortunately, the documentation pointed out that it only worked for Kotlin. Is there a setting that I can include to let Wire generate Interfaces for Java? 

If this feature is not available, do you guys know of a library that is compatible with wire to use? 

Any examples would be greatly appreciated!

Thanks in advance!

**Comment by oldergod** [comment]
Wire doesn't generate Java interfaces unfortunately. I'm not aware of any other library that would do it except gRPC.

**Reply by rndhacker** [reply]
Thanks for your early response, is there a place where I can see a sample of a gRPC java implementation using wire?

**Reply by oldergod** [reply]
Sorry I mean that gRPC can generate java interfaces, and a service generated in Kotlin with Wire should be able to interact with the java from gRPC over the wire. They can serialize/deserialize each other but the code won't be compatible in the same code base.

**Reply by rndhacker** [reply]
Do you know if this implementation is still relevant? https://github.com/square/wire/issues/441#issuecomment-250228435

**Reply by oldergod** [reply]
It was before the Gradle plugin was available. And there was never a moment in time where Wire would generate Java interface for services. In Java, we can only generate messages and enums.

**Reply by rndhacker** [reply]
Forgive my ignorance, but is it expected that developers create their own clients and services when they use java? Or is it the norm to generate those files using the gRPC plugin?

**Reply by oldergod** [reply]
It is up to them to decide what to do. At Cash App, we don't use Java anymore, everything, including all backend services, are written in Kotlin.
